src/Tasks/Common/Resources/Strings.resx

```diff
@@ -401,4 +401,12 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1070: The application configuration file must have root configuration element.</value>
     <comment>{StrBegin="NETSDK1070: "}</comment>
   </data>
-</root>
+  <data name="AppHostNotWindows" xml:space="preserve">
+    <value>NETSDK1071: Unable to use '{0}' as application host executable because it's not a Windows PE file.</value>
```

Comment:
We'll likely want to improve this UX if users are trying to build a `WinExe` output type project and specify a non-Windows RID, but that shouldn't prohibit merging this PR.

Reply:
Currently the .targets will make it so that we effectively ignore WinExe on non-Windows RIDs. Just like we do today. I didn't want to break anything, so on Windows RIDs, we now honor the OutputType=Exe (CUI)/WinExe (GUI). On non-Windows RIDs both Exe/WinExe will behave just like now - that is the same.